plugins/backstage-plugin-coder/docs/guides/sdk-recommendations.md

```diff
@@ -0,0 +1,557 @@
+- The process of wiring up auth to all properties in the hook can be confusing. The Coder SDK throws when it makes a request that isn't authenticated, but you probably want to disable those requests altogether.
+- While the `queryKey` property is less confusing overall, it is basically a `useEffect` dependency array. Dependency arrays can be confusing and hard to wire up correctly
+
+### `useCoderQuery`
+
+`useCoderQuery` is a convenience wrapper over `useQuery`, and can basically be thought of as a pre-wired combo of `useQuery`, `useCoderAuth`, and `useCoderSdk`.
+
+#### Example
+
+```tsx
+function WorkspacesList() {
+  // Pretend that we have the same useDebouncedValue from the useQuery example
+  // above
+  const [searchFilter, setSearchFilter] = useState('owner:me');
+  const debouncedSearchFilter = useDebouncedValue({
+    value: searchFilter,
+    debounceDelayMs: 500,
+  });
+
+  // Like with useQuery, the type of query.data is automatically inferred to be
+  // of type (undefined | Workspace[]) based on the return type of queryFn
+  const query = useCoderQuery({
+    queryKey: ['workspaces', debouncedSearchFilter],
+    queryFn: ({ sdk }) => sdk.getWorkspaces({ q: debouncedSearchFilter }),
+  });
+
+  return (
+    <>
+      <label>
+        Search your deployment's workspaces
+        <input
+          type="input"
+          value={searchFilter}
+          onChange={event => setSearchFilter(event.target.value)}
+        />
+      </label>
+
+      {query.error instanceof Error && (
+        <p>Encountered the following error: {query.error.message}</p>
+      )}
+
+      {query.isLoading && <p>Loading&hellip;</p>}
+
+      <ul>
+        {query.data?.map(workspace => (
+          <CustomWorkspaceListItemComponent
+            key={workspace.id}
+            workspace={workspace}
+          />
+        ))}
+      </ul>
+    </>
+  );
+}
+```
+
+#### Pros
+
+- Reduces the total number of imports needed to get query logic set up, and pre-configures values to be more fool-proof. The Coder SDK is automatically patched into Tanstack Query's `queryFn` function context.
+- Ensures that all Coder-based queries share the same query prefix, and are correctly evicted when the user unlinks their Coder account
+- Automatically prohibits some deprecated properties from Tanstack Query v4 (using one of them is treated as a type error). This means guaranteed forwards-compatibility with Tanstack Query v5
+
+#### Cons
+
+- Introduces extra overhead to ensure that queries are fully locked down when the user is not authenticated – even if they're not needed for every use case
+- Doesn't offer quite as much fine-grained control compared to `useQuery`
+- Does not simplify process of connecting Coder queries and mutations, because a `useCoderMutation` hook does not exist (yet)
+- `queryKey` is slightly more confusing, because a new value is implicitly appended to the beginning of the array
+
+## Performing mutations
```

Comment:
This seems to just be how to do mutations with React Query, I think?  It feels out of scope to me.